crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -415,6 +422,69 @@ fn handle_end_of_line_comment_around_body<'a>(
     CommentPlacement::Default(comment)
 }
 
+fn is_clause_header(node: AnyNodeRef) -> bool {
+    matches!(
+        node.kind(),
+        NodeKind::StmtIf
+            | NodeKind::StmtFor
+            | NodeKind::StmtWhile
+            | NodeKind::StmtTry
+            | NodeKind::StmtFunctionDef
+            | NodeKind::StmtWith
+            | NodeKind::ElifElseClause
+    )
+}
+
+fn handle_suppression_comment_on_clause_header<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    // Ensure the comment is an end-of-line comment
+    if !comment.line_position().is_end_of_line() {
+        return CommentPlacement::Default(comment);
+    }
+
+    // Extract the comment text
+    let comment_text = locator.slice(comment.range());
+
+    // Check if the comment is a suppression comment (e.g., `# fmt: skip`)
+    if SuppressionKind::from_comment(comment_text).is_none() {
+        return CommentPlacement::Default(comment);
+    }
+
+    // Get the preceding node (e.g., `StmtIf`, `StmtFor`, etc.)
+    let Some(preceding_node) = comment.preceding_node() else {
+        return CommentPlacement::Default(comment);
+    };
+
+    // Handle clause headers with their own nodes (if, while, for, etc.)
+    if is_clause_header(preceding_node) {
+        // Check if the comment is on the same line as the clause header
+        let header_line_start = locator.line_start(preceding_node.start());
+        let comment_line_start = locator.line_start(comment.start());
+
+        if header_line_start == comment_line_start {
+            // The comment is on the same line as the clause header, apply fmt: skip to the entire clause
+            return CommentPlacement::trailing(preceding_node, comment);
+        }
+    }
+
+    // Handle special cases like `else`, `except`, `finally` blocks that are not their own nodes
+    if let Some(parent) = comment.enclosing_parent() {
+        if is_clause_header(parent) {
+            // Apply fmt: skip if the comment is on the same line as the 'else', 'except', or 'finally'
+            let header_line_start = locator.line_start(parent.start());
+            let comment_line_start = locator.line_start(comment.start());
+
+            if header_line_start == comment_line_start {
+                return CommentPlacement::trailing(parent, comment);
```

Comment:
I'm not sure if making the comment a trailing comment is correct. Doesn't it suppress formatting for the entire node? I think the comment should be a dangling comment, see 

https://github.com/astral-sh/ruff/blob/6ac61d7b89b2ac3789c2d8ecde505f8c739c6f35/crates/ruff_python_formatter/src/comments/placement.rs#L373-L392